libHVCAENx527App/src/HVCAENx527chAio.c

```diff
@@ -159,8 +164,11 @@ write_ao( aoRecord *pior)
 	PARPROP *pp;
 
 	pp = (PARPROP *)(pior->dpvt);
-	if( pp == NULL || pp->hvchan->epicsenabled == 0)
-		return(3);
+	if( pp == NULL || pp->hvchan->epicsenabled == 0 || pp->hvchan->hvcrate->connected == 0)
+    {
+        recGblSetSevr(pior, WRITE_ALARM, INVALID_ALARM);
+		return( 3);
+    }
```

Comment:
This condition (and the severity setting) are being repeated lots of times. Refactor into a function.